services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_utils.py

```diff
@@ -36,29 +48,80 @@ async def get_volume_by_label(label: str, run_id: RunID) -> dict[str, Any]:
     async with docker_client() as docker:
         filters = {"label": [f"source={label}", f"run_id={run_id}"]}
         params = {"filters": clean_filters(filters)}
-        data = await docker._query_json(  # pylint: disable=protected-access
+        data = await docker._query_json(  # pylint: disable=protected-access  # noqa: SLF001
             "volumes", method="GET", params=params
         )
         volumes = data["Volumes"]
-        logger.debug(  # pylint: disable=logging-fstring-interpolation
-            f"volumes query for {label=} {volumes=}"
-        )
+        _logger.debug("volumes query for label=%s volumes=%s", label, volumes)
         if len(volumes) != 1:
             raise VolumeNotFoundError(label, run_id, volumes)
-        volume_details = volumes[0]
-        return volume_details  # type: ignore
+        volume_details: dict[str, Any] = volumes[0]
+        return volume_details
 
 
-async def get_running_containers_count_from_names(
+async def _get_container(
+    docker: aiodocker.Docker, container_name: str
+) -> DockerContainer | None:
+    try:
+        return await docker.containers.get(container_name)
+    except aiodocker.DockerError as e:
+        if e.status == http_status.HTTP_404_NOT_FOUND:
+            return None
+        raise
+
+
+async def _get_containers_inspect_from_names(
     container_names: list[str],
-) -> PositiveInt:
+) -> dict[str, DockerContainer | None]:
+    # NOTE: returned objects have their associated Docker client session closed
     if len(container_names) == 0:
-        return 0
+        return {}
+
+    containers_inspect: dict[str, DockerContainer | None] = {
+        x: None for x in container_names
+    }
 
     async with docker_client() as docker:
-        filters = clean_filters({"name": container_names})
-        containers = await docker.containers.list(all=True, filters=filters)
-        return len(containers)
+        docker_containers: list[DockerContainer | None] = await logged_gather(
```

Comment:
Q: if on eof them containers triggers some docker error (which could happen I guess via some one time failure). Why not use ```return_exceptions=True``` ? and then only treating the ones that return a valid value?